nexus_constructor/json/load_from_json.py

```diff
@@ -99,42 +99,57 @@ def _find_shape_information(children: List[Dict]) -> Union[Dict, None]:
     :param children: The list of dictionaries.
     :return: The shape attribute if it could be found, otherwise None.
     """
-    for item in children:
-        if item[CommonKeys.NAME] in [SHAPE_GROUP_NAME, PIXEL_SHAPE_GROUP_NAME]:
-            return item
+    try:
+        for item in children:
+            if item[CommonKeys.NAME] in [SHAPE_GROUP_NAME, PIXEL_SHAPE_GROUP_NAME]:
+                return item
+    except KeyError:
+        return None
     return None
 
 
-# Parameter group should be typed with Group but as the inheritance from base class Group
-# by StreamGroup is broken for the attribute children,
-# parameter group cannot be typed here.
-def _add_field_to_group(item: Dict, group):
-    field_type = item[CommonKeys.TYPE]
-    if (
-        field_type == NodeType.STREAM
-    ):  # streams do not have a name field so need to be dealt with differently from other fields.
-        stream = _create_stream(item)
+def _add_field_to_group(item: Dict, group: Group):
+    child: Union[Group, Link]
+    if CommonKeys.TYPE in item:
+        if item[CommonKeys.NAME] == TRANSFORMS_GROUP_NAME:
+            return
+        field_type = item[CommonKeys.TYPE]
+        child_name = item[CommonKeys.NAME]
+        if field_type == NodeType.GROUP:
+            child = _create_group(item, group)
+        elif field_type == NodeType.LINK:
+            child = _create_link(item)
+        else:
+            raise Exception(
+                f"Found unknown field type when loading JSON - {child_name}"
```

Comment:
Maybe print at the end "{child_name} is of type {field_type}" instead?

Reply:
Done.